lib/active_merchant/billing/gateways/redsys_rest.rb

```diff
@@ -378,6 +390,31 @@ def add_authentication(post, options)
         post[:DS_MERCHANT_MERCHANTCODE] = @options[:login]
       end
 
+      def add_stored_credentials(post, options)
+        return unless stored_credential = options[:stored_credential]
+
+        post[:DS_MERCHANT_COF_INI] = stored_credential[:initial_transaction] ? 'S' : 'N'
+
+        post[:DS_MERCHANT_COF_TYPE] = {
+          'recurring' => 'R',
+          'installment' => 'I'
+        }[stored_credential[:reason_type]] || 'C'
+
+        post[:DS_MERCHANT_IDENTIFIER] = 'REQUIRED' if stored_credential[:initiator] == 'cardholder'
+        post[:DS_MERCHANT_COF_TXNID] = stored_credential[:network_transaction_id] if stored_credential[:network_transaction_id].present?
```

Comment:
I don't think you need a present check
```suggestion
        post[:DS_MERCHANT_COF_TXNID] = stored_credential[:network_transaction_id] if stored_credential[:network_transaction_id]
```

Reply:
OK! corrected!